src/http.rs

```diff
@@ -53,38 +59,53 @@ pub async fn do_http_request(
         add_metric!(request_cycles_charged, cost);
         add_metric!(request_cycles_refunded, cycles_available - cost);
     }
-    inc_metric_entry!(host_requests, host);
-    let request_headers = vec![
+    inc_metric_entry!(host_requests, host.to_string());
+    let mut request_headers = vec![
         HttpHeader {
             name: "Content-Type".to_string(),
             value: "application/json".to_string(),
         },
-        HttpHeader {
-            name: "Host".to_string(),
-            value: host.to_string(),
-        },
+        // HttpHeader {
+        //     name: "Host".to_string(),
+        //     value: host.to_string(),
+        // },
     ];
+    request_headers.extend(api.headers);
     let request = CanisterHttpRequestArgument {
-        url: service_url,
+        url: api.url,
         max_response_bytes: Some(max_response_bytes),
         method: HttpMethod::POST,
         headers: request_headers,
         body: Some(json_rpc_payload.as_bytes().to_vec()),
         transform: Some(TransformContext::from_name(
-            "__transform_eth_rpc".to_string(),
+            "__transform_evm_rpc".to_string(),
             vec![],
         )),
     };
     match make_http_request(request, cost).await {
-        Ok((result,)) => {
-            String::from_utf8(result.body).map_err(|_| EthRpcError::ResponseParseError)
-        }
+        Ok((response,)) => Ok(response),
         Err((r, m)) => {
             inc_metric!(request_err_http);
-            Err(EthRpcError::HttpRequestError {
-                code: r as u32,
+            Err(HttpOutcallError::IcError {
+                code: r,
                 message: m,
```

Comment:
More informative variable names would be preferred.   :slightly_smiling_face:

Reply:
Same here--it's worth reiterating that this is based on an existing codebase, so let me know if you catch anything else like this. I promise that this is all much better than it was a few months ago.